### PR TITLE
Pin mlserver

### PR DESCRIPTION
**Pull request by serena-ruan**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/serena-ruan/mlflow/pull/11280?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/11280/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 11280
```

</p>
</details>

### Related Issues/PRs

 #xxx

### How is this PR tested?

- [x] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests

### Does this PR require documentation update?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/models/container/__init__.py [hunk 1]
```diff
@@ -43,6 +46,10 @@
 
 SERVING_ENVIRONMENT = "SERVING_ENVIRONMENT"
 
+# pydantic 2.6.3 has breaking changes that affect the mlflow server
+# TODO: remove this after https://github.com/SeldonIO/MLServer/pull/1595 is applied
+PYDANTIC_REQUIREMENT = "pydantic<2.6.3"
```
Comment (serena-ruan) [inline mlflow/models/container/__init__.py:51]: core change

File: tests/helper_functions.py [hunk 2]
```diff
@@ -33,6 +33,9 @@
 AWS_METADATA_IP = "169.254.169.254"  # Used to fetch AWS Instance and User metadata.
 LOCALHOST = "127.0.0.1"
 PROTOBUF_REQUIREMENT = "protobuf<4.0.0"
+# pydantic 2.6.3 has breaking changes that affect the mlflow server
+# TODO: remove this after https://github.com/SeldonIO/MLServer/pull/1595 is applied
+PYDANTIC_REQUIREMENT = "pydantic<2.6.3"
```
Comment (serena-ruan) [inline tests/helper_functions.py:38]: +1

Comment (daniellok-db) [reply tests/helper_functions.py:38]: is it possible to import this from `mlflow.models.container` instead? in case we forget to update one

File: tests/helper_functions.py [hunk 3]
```diff
@@ -237,7 +237,12 @@ def pyfunc_serve_and_score_model(
         validate_version = "--enable-mlserver" not in extra_args
     proc = _start_scoring_proc(cmd=scoring_cmd, env=env, stdout=stdout, stderr=stdout)
     return _evaluate_scoring_proc(
-        proc, port, data, content_type, activity_polling_timeout_seconds, validate_version
+        proc,
+        port,
+        data,
+        content_type,
+        activity_polling_timeout_seconds,
+        validate_version,
```
Comment (harupy) [inline tests/helper_functions.py:245]: can you revert unrelated changes?

**Review by harupy — APPROVED** [review]
LGTM!
